src/utils/auth.ts

```diff
@@ -23,7 +39,7 @@ const fetchNonce = async () => {
 export const login = async (
   chainId: number,
   address: string,
-  signFunction: (args?: any) => Promise<`0x${string}`>
+  signFunction: (args?: any) => Promise<`0x${string}` | undefined>
```

Comment:
What's the scenario that the sign function would return `undefined`? And if it does, we probably need to handle that scenario in this `login` function.

Reply:
Good catch. There is no scenario. I was planning other solution first but realised it is better to just let sigFunction throw if, for instance, user rejects signature. Removing.